state/core_access.go

```diff
@@ -203,6 +194,14 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		return nil, fmt.Errorf("account for signer %s not found", author)
 	}
 
+	gas := cfg.GasLimit()
+	if gas == 0 {
+		gas, err = ca.estimateGasForBlobs(account.Address().String(), libBlobs)
```

Comment:
this path looks like it gets overwritten anyway in `BroadcastPayForBlobWithAccount` -- which is interesting because AFAIU it would also write any custom value set in txconfig..

Reply:
It would be nice if we had some sort of integration test that would ensure that values set in config were actually the values with which txclient builds + signs the tx.

Reply:
Their gas estimation is prepended to the passed options. So, our values will overwrite their.